File: src/py-opentimelineio/opentimelineio/schema/stream_mapper.py
# SPDX-License-Identifier: Apache-2.0
# Copyright Contributors to the OpenTimelineIO project

from .. core._core_utils import add_method
from .. import _otio


@add_method(_otio.StreamMapper)
def __str__(self):
    return (
        "StreamMapper("
        "{}, "
        "{}"
        ")".format(
            str(self.name),
            str(self.stream_map),
        )
    )


@add_method(_otio.StreamMapper)
def __repr__(self):
    return (
        "otio.schema.StreamMapper("
        "name={}, "
        "stream_map={}"
        ")".format(
            repr(self.name),
            repr(self.stream_map),
        )
    )

File: src/opentimelineio/streamMapper.cpp
// SPDX-License-Identifier: Apache-2.0
// Copyright Contributors to the OpenTimelineIO project

#include "opentimelineio/streamMapper.h"

namespace opentimelineio { namespace OPENTIMELINEIO_VERSION_NS {

StreamMapper::StreamMapper(
    std::string const&   name,
    std::string const&   effect_name,
    StreamMap const&     stream_map,
    AnyDictionary const& metadata)
    : Parent(name, effect_name, metadata)
    , _stream_map(stream_map)
{}

StreamMapper::~StreamMapper()
{}

bool
StreamMapper::read_from(Reader& reader)
{
    return reader.read_if_present("stream_map", &_stream_map)
           && Parent::read_from(reader);
}

void
StreamMapper::write_to(Writer& writer) const
{
    Parent::write_to(writer);
    writer.write("stream_map", _stream_map);
}

}} // namespace opentimelineio::OPENTIMELINEIO_VERSION_NS
